app/Http/Controllers/HomeController.php

```diff
@@ -40,4 +42,36 @@ public function show($id)
 
         return view('otheruser',compact('user'));
     }
+
+    public function edit($user)
+    {
+        $user = auth()->user();
+
+        return view('editprofile',compact('user'));
+    }
+
+    public function editprofile_update(Request $request)
+    {
+        $this->validate($request,[
+            'name'=> 'required|min:3|max:50|',
+            'email'=>'required|min:3|max:50|',
+        ]);
+
+        if($request->newpassword != null){
+            $request->validate([
+                'newpassword'=> ['required', 'string', 'min:6', 'confirmed']
+            ]);
+            Auth::user()->update([
+                'password' => bcrypt($request->newpassword)
+                ]);
+        
+        }    
+        
+        Auth::user([
+            'name' => $request->name,
+            'email' => $request->email,
+        ]);
```

Comment:
I think you need this for it to work.
```suggestion
        Auth::user()->update([
            'name' => $request->name,
            'email' => $request->email,
        ]);
```